pylint/checkers/refactoring/implicit_booleaness_checker.py

```diff
@@ -166,13 +162,15 @@ def _check_use_implicit_booleaness_not_comparison(
                 literal_node = comparator if is_right_empty_literal else node.left
                 # Infer node to check
                 try:
-                    target_instance = next(target_node.infer())
+                    target_instance = utils.safe_infer(target_node)
                 except astroid.InferenceError:
                     # Probably undefined-variable, continue with check
                     continue
+                if not target_instance:
+                    continue
                 mother_classes = self.base_classes_of_node(target_instance)
                 is_base_comprehension_type = any(
-                    t in mother_classes for t in ("tuple", "list", "dict")
+                    t in mother_classes for t in ("tuple", "list", "dict", "set")
```

Comment:
> Did we not activate the check we have for that ?

I also confused that we seem not activate the check for the use-implicit-booleaness-not-comparison (C1803)

Reply:
@zjzh are you using the latest pylint from the dev branch ? Because it's not yet released (it will be in 2.12). You can use ``pip install git+git://github.com/PyCQA/pylint.git@main#egg=pylint -U``